lua/lualine/highlight.lua

```diff
@@ -1,24 +1,58 @@
 local M = {  }
 
-local function highlight (name, foreground, background, special)
-  local command = {
-    'highlight', name,
-    'guifg=' .. foreground,
-    'guibg=' .. background,
-    'gui=' .. (special or 'none'),
-  }
+local function get_cterm_color(color)
+  local function round(num)
+    if num - math.floor(num) < 0.5000000 then
+      return math.floor(num)
+    else
+      return math.ceil(num)
+    end
+  end
+
+  local r = tonumber(color:sub(2,3), 16)
+  local g = tonumber(color:sub(4,5), 16)
+  local b = tonumber(color:sub(6,7), 16)
+  return round(36 * round(r * 5/256) + 
+               6 * round(g * 5/256) +
+               round(b * 5 / 256) + 16) 
+end
+
+local function highlight (name, foreground, background, gui)
+  local command = {}
+  if type(foreground) == 'table' and type(background) == 'table' then
```

Comment:
This is a mess, here is how to do it better
```lua
local function highlight (name, foreground, background, gui)
  local command = {
    'highlight', name,
    'ctermfg=' .. (foreground[2] or get_cterm_color(foreground)),
    'ctermbg=' .. (background[2] or get_cterm_color(background)),
    'cterm=' .. (gui or 'none'),
    'guifg=' .. (foreground[1] or foreground),
    'guibg=' .. (background[1] or background),
    'gui=' .. (gui or 'none'),
  }
  return table.concat(command, ' ')
end
```

Reply:
Agreed